src/sssom/util.py

```diff
@@ -246,7 +246,11 @@ def clean_prefix_map(self, strict: bool = True) -> None:
         if self.metadata:
             prefixes_in_table.update(get_prefixes_used_in_metadata(self.metadata))
 
-        missing_prefixes = prefixes_in_table - self.converter.get_prefixes()
+        prefixes = {record.prefix for record in self.converter.records}
+        aliases = {p for p in itt.chain(*[record.prefix_synonyms for record in self.converter.records]) if p}
+        converter_prefixes = prefixes | aliases
```

Comment:
## Updates: `clean_prefix_map()`
<details><summary>Details</summary>
<p>

## Overview
I was getting this error thrown near the end of `clean_prefix_map()`:
```
ValueError: {'oio'} are used in the SSSOM mapping set but it does not exist in the prefix map
```
Basically, there were some rows in the mapping set which used `oio`, but `oio` was not detected in the prefix map.

The cause of this bug was that when it checked for prefixes in the prefix map, it was only checking `prefix` on each `Record` and ignoring `prefix_synonyms`.

## Alternative solutions
I was considering that instead of doing this fix in `sssom-py`, a fix could also be implemented in `curies`. The bug I described above is actually exhibited in `curies` in `Converter.get_prefixes()`, so the same fix could be implemented there. Perhaps it could also be parameterized like, `Converter.get_prefixes(include_synonyms=True)`.

</p>
</details>

Reply:
Why are there two seperate prefixes for the same thing your mapping set?

Reply:
also, this is not related to the other changes so it would make much more sense to break this up into 2 PRs

Reply:
@matentzn I'm not sure how `oio` and `oboInOwl` got into `mondo-ingest`'s `mondo.sssom.config.yaml`; I mostly don't use that. They have duplicative URI prefixes.
However even when removing 1 of them from `mondo.sssom.config.yaml`, the problem arises again. Basically the lexmatch pipeline gets and merges 2 prefix maps. One of them comes from `mondo.sssom.config.yaml`. The second is dynamically generated by reading the contents of Mondo.

Reply:
I understand that they're both motivated by the same work that you're doing, but they are not related code changes, which makes working through this PR not easy nor enjoyable. This will also slow down the speed we're able to work through this

Reply:
There's not a lot of code changes here. I'm surprised if this is a difficult PR to work through.

@matentzn @cthoyt But yes there are 2 different issues here, so I can split into 2 different PRs if you guys want?
1. PR for dealing with bijective prefix maps (`convert_plain_prefix_map_to_extended()` and `get_metadata_and_prefix_map()`
2. PR for what I think is a bug in `clean_prefix_map()` where it does not detect prefix synonyms.

Reply:
This fix has been superseded with `curies` solution which does the same things. See [this comment](https://github.com/mapping-commons/sssom-py/pull/485#discussion_r1455360522).